front/lib/resources/agent_message_feedback_resource.ts

```diff
@@ -43,130 +61,357 @@ export class AgentMessageFeedbackResource extends BaseResource<AgentMessageFeedb
     );
   }
 
-  static async listByAgentConfigurationId({
-    agentConfiguration,
-  }: {
-    agentConfiguration: AgentConfigurationType;
-  }): Promise<AgentMessageFeedbackResource[] | null> {
-    const agentMessageFeedback = await AgentMessageFeedback.findAll({
-      where: {
-        agentConfigurationId: agentConfiguration.sId,
-      },
-      order: [["id", "DESC"]],
-    });
+  async delete(
+    auth: Authenticator,
+    { transaction }: { transaction?: Transaction } = {}
+  ): Promise<Result<undefined, Error>> {
+    try {
+      await this.model.destroy({
+        where: {
+          id: this.id,
+        },
+        transaction,
+      });
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
 
-    return agentMessageFeedback.map(
-      (feedback) => new this(this.model, feedback.get())
-    );
+  async updateFields(
+    blob: Partial<
+      Pick<
+        AgentMessageFeedback,
+        "content" | "thumbDirection" | "isConversationShared"
+      >
+    >
+  ) {
+    return this.update({
+      content: blob.content,
+      thumbDirection: blob.thumbDirection,
+      isConversationShared: blob.isConversationShared,
+    });
   }
 
-  static async fetchByUserAndAgentMessage({
-    user,
-    agentMessage,
+  static async fetch({
+    workspaceId,
+    withMetadata,
+    agentConfiguration,
+    filters,
   }: {
-    user: UserType;
-    agentMessage: AgentMessage;
-  }): Promise<AgentMessageFeedbackResource | null> {
-    const agentMessageFeedback = await AgentMessageFeedback.findOne({
+    workspaceId: number;
```

Comment:
number should be ModelId but we def don't want ModelId in our interfaces. ModelId like Sequelize Models should be constrained inside of resource and not leak outside.

Let's use WorkspaceType here

Reply:
I ended up using the workspaceId to avoid refactoring [workspace usage ](https://github.com/dust-tt/dust/blob/main/front/lib/workspace_usage.ts#L462) and go faster.
If you're strongly opinionated I can refactor workspace usage too.